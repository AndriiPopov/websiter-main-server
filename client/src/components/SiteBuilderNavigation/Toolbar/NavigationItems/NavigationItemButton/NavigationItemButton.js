import React from 'react';
//import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import classes from './NavigationItemButton.module.css';
//import * as actions from '../../../../store/actions/index';
import Svg from '../../../../Svg/Svg';

export const NavigationItemButton = props => {
    const addedClasses = [ classes.NavigationItemButton ];
    if (props.disabled) {
        addedClasses.push(classes.disabled);
    }
    if (props.active) {
        addedClasses.push(classes.active);
    }

    return (
        <li 
            className={addedClasses.join(' ')}
            onClick={props.disabled ? null : props.clicked}>
            <Svg icon={props.icon} />
        </li>
    );
};

NavigationItemButton.propTypes = {
    icon: PropTypes.string,
    disabled: PropTypes.bool,
    clicked: PropTypes.func
};

export default NavigationItemButton;