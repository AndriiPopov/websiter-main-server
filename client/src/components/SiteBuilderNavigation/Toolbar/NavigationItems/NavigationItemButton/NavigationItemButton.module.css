.NavigationItemButton {
    color: #555;
    cursor: pointer;
    padding: 15px 10px;
    height: 100%;
    vertical-align: bottom;
    fill: #555;
    margin: 0px;
    display: flex;
    width: auto;
    align-items: center;
    box-sizing: border-box;
}

.NavigationItemButton:hover,
.NavigationItemButton:active,
.NavigationItemButton.active {
    background: #eee;
}

.NavigationItemButton a.disabled {
        color: grey;
}
