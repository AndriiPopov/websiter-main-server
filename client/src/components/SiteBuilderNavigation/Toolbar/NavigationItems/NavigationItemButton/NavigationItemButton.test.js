import React from 'react';
import {shallow, configure} from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

import { NavigationItemButton } from './NavigationItemButton';

configure({adapter: new Adapter()});

describe('NavigationItemButton', () => {
    it('should be defined', () => {
        expect(NavigationItemButton).toBeDefined();
    });

    it('it should render button with onclick null when disabled', () => {
        const tree = shallow(
            <NavigationItemButton
                disabled
                clicked="clicked"
                icon="<svg></svg>" />
        );
        expect(toJson(tree)).toMatchSnapshot();
    });

    it('it should render button with onclick null when disabled', () => {
        const tree = shallow(
            <NavigationItemButton
                clicked="clicked"
                icon="<svg></svg>" />
        );
        expect(toJson(tree)).toMatchSnapshot();
    });
});