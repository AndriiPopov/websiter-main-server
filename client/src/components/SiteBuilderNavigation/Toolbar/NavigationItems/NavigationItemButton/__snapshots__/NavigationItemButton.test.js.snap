// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationItemButton it should render button with onclick null when disabled 1`] = `
<li
  className="NavigationItemButton disabled"
  onClick={null}
>
  <Svg
    icon="<svg></svg>"
  />
</li>
`;

exports[`NavigationItemButton it should render button with onclick null when disabled 2`] = `
<li
  className="NavigationItemButton"
  onClick="clicked"
>
  <Svg
    icon="<svg></svg>"
  />
</li>
`;
